### ISSUE TITLE
Images for product variants

### ISSUE THREAD

**Issue by oklinge** [MicrosoftDocs/dynamics-365-unified-operations-public#3495]
The documentation is not correct. I submitted a service request (2107080050001473) on this issue, it is not possible to add dimension specific images to pictures. And have the pictures change upon selection change in the web shop.

Furthermore the documentation states that dimension specific images only works with Style, Size and Color, this is also not true. As I have added images for configuration. Actually only managed to add configuration specific images and not on any of the stated dimensions (Style, Size and Color). To do this I used this filename format:

[Item number] ^ [Configuration Name]_000_001.png

Example:
123ABC ^ Yellow_000_001.png

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 6d8c889e-fd24-912c-61bd-40abaa7a18ca
* Version Independent ID: 79ac40fe-2a63-07f0-24ac-4c61d56d25ef
* Content: [Upload images - Commerce | Dynamics 365](https://docs.microsoft.com/en-us/dynamics365/commerce/dam-upload-images)
* Content Source: [articles/commerce/dam-upload-images.md](https://github.com/MicrosoftDocs/Dynamics-365-Unified-Operations-Public/blob/main/articles/commerce/dam-upload-images.md)
* Service: **dynamics-365-commerce**
* Product: ****
* Technology: ****
* GitHub Login: @psimolin
* Microsoft Alias: **psimolin**

**Comment by oklinge** [comment]
Today I succeeded in adding dimension specific images on items. Under is the procedure I did to find the correct format in which to name the product images.

Procedure in which I found the format to use:

1. Went to a product with color dimension
2. Pressed F12, then console
3. Changed the color of the item, watched lots of errors emerging in the console pane
4. Copied link address of one of the errors that looked like a search for an image
![image](https://user-images.githubusercontent.com/80251989/128337727-4353574a-dfd0-4e0a-9f67-a3e3afddbbd6.png)

5. Went and decoded the HTML encoded URL and found the image name it was looking for, in this case this was «100008 ^  ^ 30 ^ Navy ^  ^_000_001». Here is a representation of the string displaying all characters: 
![image](https://user-images.githubusercontent.com/80251989/128337905-7147f6b9-b741-4284-8cf4-d3582d9f4f16.png)
6. Renamed an image exactly as was searched for, «100008 ^  ^ 30 ^ Navy ^  ^_000_001.png» and uploaded and published to the portal. 

This worked and we are now successful in displaying dimension specific images on products in the web portal.

The information displayed in the article **Upload images** [https://docs.microsoft.com/en-us/dynamics365/commerce/dam-upload-images](https://docs.microsoft.com/en-us/dynamics365/commerce/dam-upload-images) is misleading. This because the actual format provided is not correct ("/Products/{ProductNumber} ^ {Style} ^ {Size} ^ {Color} ^_000_001.png"). And the example is missing whitespaces and a special character ("93039 ^ ^ 2 ^ Black ^_000_001.png"). You can see this by displaying all the characters and comparing it to the format that I found working. The example taken from the article looks like this:
![image](https://user-images.githubusercontent.com/80251989/128336605-8ece6db1-59ec-4bb8-9e3c-3b8b5259e215.png)
Missing a vital whitespace in between first set of the special characters and an additional special character with two whitespaces at the end before the postfix. 

Please fix the documentation and display an example that is correct. It should be like this :
![image](https://user-images.githubusercontent.com/80251989/128339045-82ec86d0-a0f8-4114-8874-8e8a25410462.png)

**Comment by RobinARH** [comment]
Created bug 610093 to track this issue.

**Comment by psimolin** [comment]
Hello @oklinge,

Thanks for detailed bug report. I have created a PR with the whitespace fix in the example. The missing caret ("additional special character") I was not able to confirm using the default demo data configuration.

I added a note to the documentation, that the naming conventions shown for product variant images are examples using default configuration for media naming and dimensions. As you know, both of which are configurable in the product.
